File: src/main/java/org/taktik/icure/logic/FrontEndMigrationLogic.java
package org.taktik.icure.logic;

import org.taktik.icure.entities.FrontEndMigration;
import org.taktik.icure.exceptions.DeletionException;

import java.util.List;

public interface FrontEndMigrationLogic extends EntityPersister<FrontEndMigration, String>{

    FrontEndMigration createFrontEndMigration(FrontEndMigration frontEndMigration);

    String deleteFrontEndMigration(String frontEndMigrationId) throws DeletionException;

    FrontEndMigration getFrontEndMigration(String frontEndMigrationId);

    List<FrontEndMigration> getFrontEndMigrationByUserIdName(String userId, String name);

    FrontEndMigration modifyFrontEndMigration(FrontEndMigration frontEndMigration);
}

File: src/main/kotlin/org/taktik/icure/asynclogic/FrontEndMigrationLogic.kt
package org.taktik.icure.asynclogic

import kotlinx.coroutines.flow.Flow
import org.taktik.couchdb.DocIdentifier
import org.taktik.icure.asyncdao.FrontEndMigrationDAO
import org.taktik.icure.entities.FrontEndMigration

interface FrontEndMigrationLogic : EntityPersister<FrontEndMigration, String> {
    suspend fun createFrontEndMigration(frontEndMigration: FrontEndMigration): FrontEndMigration?
    suspend fun deleteFrontEndMigration(frontEndMigrationId: String): DocIdentifier?

    suspend fun getFrontEndMigration(frontEndMigrationId: String): FrontEndMigration?
    fun getFrontEndMigrationByUserIdName(userId: String, name: String?): Flow<FrontEndMigration>

    suspend fun modifyFrontEndMigration(frontEndMigration: FrontEndMigration): FrontEndMigration?
    fun getGenericDAO(): FrontEndMigrationDAO
}
